src/Compilers/Test/Core/Assert/ConditionalFactAttribute.cs

```diff
@@ -332,6 +333,12 @@ public class ClrOnly : ExecutionCondition
         public override string SkipReason => "Test not supported on Mono";
     }
 
+    public class NotMono : ExecutionCondition
```

Comment:
Please make the naming a more crisp here to denote if it's any, desktop or core mono.

Reply:
How about using "NotOnMonoCore" ?